modules/references/server/controllers/reference.server.controller.js

```diff
@@ -375,41 +375,48 @@ exports.readMany = async function readMany(req, res, next) {
     // validate the query
     validate(validateReadMany, req);
 
-    const { userFrom, userTo } = req.query;
+    const { userTo, userFromPublic, userFromPending } = req.query;
```

Comment:
Can the `userFromPending` be inferred from authenticated user `(available always as `req.user._id`), and it thus wouldn't need to be passed through from client layers?

Reply:
To elaborate more on this; we should surface pending references only to people who wrote them, i.e. the authenticated user.

Reply:
> Could the userFromPending be inferred from the authenticated user (available always as req.user._id), and it thus wouldn't need to be passed through from client layers?

Right, very valid point.

In general, there so many changes that depend on the change of the experience fetching API. I'll look thoroughly look into the question of changing API in the context of all tasks.